dali/pipeline/operators/crop/new_crop.cc

```diff
@@ -34,21 +35,32 @@ DALI_SCHEMA(NewCrop)
 
 namespace detail {
 
-template <typename InputType, typename OutputType, std::size_t D>
+template <typename InputType, typename OutputType>
 void RunHelper(Tensor<CPUBackend> &output,
                const Tensor<CPUBackend> &input,
-               const std::array<int64_t, D> &slice_anchor,
-               const std::array<int64_t, D> &slice_shape) {
-  kernels::KernelContext ctx;
-  auto in_view = view<const InputType, D>(input);
-
-  kernels::SliceArgs<D> slice_args = {slice_anchor, slice_shape};
-
-  kernels::SliceCPU<OutputType, InputType, D> kernel;
-  kernels::KernelRequirements kernel_req = kernel.Setup(ctx, in_view, slice_args);
-
-  auto out_view = view<OutputType, D>(output);
-  kernel.Run(ctx, out_view, in_view, slice_args);
+               const std::vector<int64_t> &slice_anchor,
+               const std::vector<int64_t> &slice_shape) {
+  std::size_t number_of_dims = input.shape().size();
+  VALUE_SWITCH(number_of_dims, NumDims, (3, 4), (
+    kernels::KernelContext ctx;
+    auto in_view = view<const InputType, NumDims>(input);
+
+    std::array<int64_t, NumDims> anchor, shape;
+    for (std::size_t d = 0; d < NumDims; d++) {
+      anchor[d] = slice_anchor[d];
+      shape[d] = slice_shape[d];
+    }
+    kernels::SliceArgs<NumDims> slice_args = {std::move(anchor), std::move(shape)};
```

Comment:
What are those moves supposed to do? 

You could create empty SliceArgs, and fill it directly as in loop above.

Reply:
ok, I'll change that